### ISSUE TITLE
A UserAgent header már nem kell.

### ISSUE THREAD

**Issue by thegergo02** [boapps/e-kreta-api-docs#19]
Terveztem átírni a librarym, de perfektül megkaptam a számonkéréseket ma a header nélkül. A felhasználó adatokat mondjuk még nem teszteltem, de a számonkérések nem mentek tegnap.

**Comment by boapps** [comment]
azért a biztonság kedvéért ott hagyom még pár napig, aztân meglátjuk

**Comment by thegergo02** [comment]
Oké, ma nagyon furán megy. A User-Agent megint kell, viszont sokszor megint "connection reset by peer" még a header-el is. De a legrosszabb hogy nem mindig ugyanaz van.

**Comment by thegergo02** [comment]
Már nem tudok másra gondolni csak arra hogy direkt csinálják ezt.

**Comment by thegergo02** [comment]
Próbáltam a tesztek közé egy 10mp-s sleepet rakni, semmi változás tehát nem rate limitm

**Comment by thegergo02** [comment]
Legtöbbször a dátum nélküli requestek failelnek(?)

**Comment by boapps** [comment]
Ezzel a headerrel még most is megy: "User-Agent: Kreta.Ellenorzo/2.9.3.2019100701 (Android; coral 0.0)"

Gondolom látták, hogy megoldottuk, ezért elkezdték szigorúbban blokkolni (eddig csak az elejét nézték). Ez az új user agent headerünk megkülönbözhetetlen az övéktől. Esetleg blacklistelhetik ezt a headert is, akkor viszont legit felhasználókat is blokkolnának, ennyit csak nem ér meg nekik, mi meg még mindig beállíthatunk egy random stringet a codename helyére.

Nem értem őket, ez miért jó? Azt hiszik, nem tudjuk megváltoztatni a headert vagy nem jövünk rá, hogy kell, vagy csak arra az egy napra játszanak, hogy ne működjön a szivacs és lehessen rá mutogatni, hogy na mégis a miénk a jobb? Ez szánalmas, nem felháborító egyszerűen csak szánalmas. Csak magatokat járatjátok le!

**Comment by thegergo02** [comment]
Egyetértek, ha rosszul esik hogy egy harmadik féltől származó app jobban müködik mással is lehetne foglalkozni. Amúgy köszi a headert, remélem holnap kitudom próbálni.

**Comment by palmarci** [comment]
a user agent mezo kotelezonek nez ki, a kretasok kitalaltak h a .net vagy c# vagy miaz binarisba hardcodeoljak az /Core/Models/Environments/EnvironmentSettings/EnvironmentSetting.cs file DefaultUserAgentHeader értéket az Kreta.Ellenorzo/[google play verzioszam]/[apk verzio kod]-ra

az utana levo coral-os stringrol nem tudok semmit, inkabb benne hagytam

hozzaraktam a scriptbe ami orankent lefrissiti a github mirrorunkat (amiben mar nincs benne a reklam se), azt nyugodtan lehet hasznalni referenciakent

https://github.com/boapps/kreta-api-mirror/blob/master/user-agent

**Comment by palmarci** [comment]
remelem olvassak a kretasok es tudjak h hova kuldenem el oket

**Comment by thegergo02** [comment]
Ez nagyon jól jön mindenkinek, köszi!

**Comment by boapps** [comment]
a coral helyén lenne az aktuális telefon codename. A coral, az a pixel 4 codeneve, szóval ha ezt blokkolják, akkor azzal a blokkolják saját felhsználóikat is. Én nem akarom elküldeni a krétásokbak, hogy a feljasználóim milyen telefont használnak, esetleg random stringet lehetne beleírni hogyha így is blokkolnak.

**Comment by palmarci** [comment]
logikus, majd kitalalunk valamit

**Comment by antai1988** [comment]
nos boa érdekes amit csinálsz, az alkalmazásod állítja hogy ő a hivatalos állami kréta app, használod az API kulcsát, publikálod... vagy nagyon bátor vagy nagyon ostoba :-). azért légy óvatos mert lehet amit te a krétások bénázásának vélsz, azzal téged tesztelnek, gyűjtik az infokat... inkább használj saját API kulcsot és a U.A -ban is korrektül azonosítsd magad.

**Comment by boapps** [comment]
@quantumLeap1988 Az api kulcs az egy dolog lehet, az amit kiraktam, és nem titok, bárki (ahogy írtam is) meg tudja nézni a kréta lekérdezéseiből, mást egyébként se tudnék használni. Ezt az api kulcsot te szerintem kevered a user agent headerrel, aminél szintén az az egyetlen amit beküldtem és szintén lekérdezésben látszik és szintén nem tudok mást használni, mert akkor nem engedi a lekérdezést és kapunk egy tök korrektül azonsító appot, ami semmit nem tud csinálni! A kréta amúgy engedi elvileg a nem hivatalos appok működését, másképp meg nem tudok ilyen appot csinálni.

**Comment by boapps** [comment]
az alkalmazásod állítja hogy ő a hivatalos állami kréta app ez pedig egyszerűen nem igaz

**Comment by szaboand** [comment]
ha egy másik app kulcsát használod az nem okés gondold el mi lenn egy banki alkalmazás esetén ha a hivatalos app kulcsát kezdené használni. ugyanez igaz a UA headerre is. a tiltástól meg ne aggódj ha tiltani akarnának akkor arra komolyabb eszközök vannak amit tudsz megkerülni. minden API term of use leírja a használat feltételeit, érdemes elolvasnod. azzal nem védekezhetsz hogy azért csinálod illegálba mert másképp nem tudod. szóval hajrá, de azért csak okosan

**Comment by quantumLeap1988** [comment]
az API is jogvédett dolog olvasd el a API term of use-t. abban le van írva hogy mit tehetsz és mit nem. az API keyt biztos nem másolhatod azt minden API fejlesztő tilt. ez legal dolog nem technikai, azzal nem védkezhetsz hogy másképp nem tudod megcsinálni. ha védeni akarják az apit arra hard cuccok is vannak amit nem tudsz megkerülni. ha én nem zárom a kapumat az még nem azt jelenti hogy bejöhetsz.

**Comment by pepyta** [comment]
Tényleg jó ötletnek tűnne a Kréta ügyfélszolgálatát emiatt megkeresni, de az index írt az eSzivacsról egy cikket, amiben a KRÉTA munkatársa véleménye is szerepelt (https://index.hu/belfold/2019/09/25/oktatas_e-naplo_kreta_rendszer_adathalaszat_telefonos_alkalmazas_fejleszto/), tehát gondolom tudnak a dologról.

**Comment by quantumLeap1988** [comment]
petya a GMap, FB is igényelni kell keyt. kréta is belinkelheti az issuet hogy tudtok róla. 
**ignorantia non excusat legem**.

**Comment by boapps** [comment]
> olvasd el a API term of use
Na ezt szívesen elolvasom, hol van?

**Comment by boapps** [comment]
Egyébként feltételezem, hogy ha bajuk lenne az appommal, akkor már írtak volna.

**Comment by thegergo02** [comment]
Nyugodtan csinálhatnak egy takedownt ha zavarja őket valami. Lásd: https://www.crunchyroll.com/forumtopic-1021185/crunchyroll-plugin-for-kodi-dmca-takedown

**Comment by thegergo02** [comment]
A szituáció majdnem ugyanaz.

**Comment by thegergo02** [comment]
Mondjuk én (vagy akár mindenki) örülne egy hivatalos API key szerzési lehetőséghez, akkor nem kéne hamis requesteket csinálnunk...

**Comment by thegergo02** [comment]
@quantumLeap1988 Ha már latin mondatok is használunk, nekem is van egy erre: **Leges instituuntur cum promulgantur**

**Comment by boapps** [comment]
Én egyébként már írtam a krétásoknak a useragentes blokkolások miatt, remélem közösen kitalálunk egy olyan megoldást, ami mindenkinek jó, mert van rá az esély, hogy nem azért blokkolták a szivacsot mert rontani akarják a hírnevét. Lehet valami bajuk van azzal, hogy az app túl sok lekérdezést küld, vagy nem elég világos, hogy nem hivatalos (bár ezzel én nem értek egyet) vagy valami hasonló. Mondjuk ilyen esetben is illene előbb szólni. 

Azzal nem lenne problémám, ha lenne külön api kulcs, de az a baj, hogy nincs, azzal se lenne problémám, ha külön UserAgentet kéne használnom, de akkor nem működne az app, mert mint mondtam, azzal nem megy a lekérdezés. Mindegy, én várom a válaszukat, de addig nem hagyom, hogy blokkoljanak, amíg nincs rá indokuk, vagy hivatalosan nem tiltják meg nekem a fejlesztést. Ha írnának, hogy jólvan ezt fejezd be, akkor én abbahagynám a szivacs fejlesztését, mert szerintem onnantól lenne illegális, de ilyen sem volt, azt írták nyugodtan csináljam, még bevételt is szerezhetek belőle (off: opcionális, alapból kikapcsolt reklámok cooming soon a playes verzióba és lesz egy külön ad-free foss F-droid/github release).

**Comment by quantumLeap1988** [comment]
**Rosszul teszed, ha dicsérsz olyan dolgot, amit nem értesz igazán, de még rosszabbul azt, hogyha szidod.** LdaV. Peace! ☮

**Comment by boapps** [comment]
A kréta visszaírt, többek között nem engedik a UserAgentet és az apiKey használatát, jogi hülyeségekre hivatkozva, ezeket most szedjétek ki ti is az appotokból, mert szerintük illegális. Ja és a név nem utalhat az e-krétára (e-szivacs is túlságosan hasonlít).

**Comment by palmarci** [comment]
ez van

**Comment by thegergo02** [comment]
Uh, mondanék pár sort szívesen most. Alternatív key szerzési lehetőség legálisan szóba se jött?
